unlock-wordpress-plugin/inc/classes/fullpostpage/class-unlock-box-fullpp.php

```diff
@@ -0,0 +1,196 @@
+ *
+ * @since 3.0.0
+ *
+ * @package unlock-protocol
+ */
+
+ namespace Unlock_Protocol\Inc\Fullpostpage;
+
+use Unlock_Protocol\Inc\Traits\Singleton;
+use Unlock_Protocol\Inc\Unlock;
+
+/**
+ * Class Unlock_Box_Fullpp
+ *
+ * @since 3.0.0
+ */
+class Unlock_Box_Fullpp {
+
+	use Singleton;
+
+	/**
+	 * Construct method.
+	 *
+	 * @since 3.0.0
+	 */
+	protected function __construct() {
+
+		$this->setup_hooks();
+
+	}
+
+	/**
+	 * Setup hooks.
+	 *
+	 * @since 3.0.0
+	 */
+	protected function setup_hooks() {
+		/**
+		 * Actions.
+		 */
+		add_filter('the_content', array( $this, 'unlock_box_fullpp_filter' ));
+
+	}
+
+	/**
+	 * Filter the content of locked posts/pages.
+	 *
+	 * @param string $content Post/Page Content.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @return string HTML elements.
+	 */
+	public function unlock_box_fullpp_filter($content) {
+		global $post;
+
+		// Step 1: Retrieve the attributes from the database
+		$attributes = get_post_meta($post->ID, 'unlock_box_attributes', true);
+
+		// Step 2: Retrieve the current up-to-date network list from the database
+		$networkList = get_option('unlock_protocol_settings');
+		$networks = $networkList['networks'];
+
+		// Step 3: Iterate through the network list to locate the corresponding network ID
+		$formattedNetworks = array_map(function ($net) {
+			return array(
+				'name' => $net['network_name'],
+				'value' => $net['network_id'],
+			);
+		}, array_values($networks));
+
+		// Check if the attributes are valid
+		if (!$attributes || !isset($attributes['locks']) || !isset($attributes['ethereumNetworks'])) {
+			return $content;
+		}
+
+
+		// Step 4: Reconstruct a new attributes format to match the old structure
+		$locks = array();
+		foreach ($attributes['locks'] as $index => $lock) {
+			$network_id = $formattedNetworks[array_search($attributes['ethereumNetworks'][$index], array_column($formattedNetworks, 'value'))]['value'];
+			$locks[] = array(
+				'address' => $lock['lockAddress'],
+				'network' => $network_id,
+			);
+		}
+
+		// Pass the reconstructed attributes to the render_fullpp_content function
+		$new_attributes = array(
+			'locks' => $locks,
+			'ethereumNetworks' => $formattedNetworks,
+		);
+		return $this->render_fullpp_content($new_attributes, $content);
+		
+	}
+
+	/**
+	 * Render block.
+	 *
+	 * @param array  $attributes List of attributes passed in block.
+	 * @param string $content Block Content.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @return string HTML elements.
+	 */
+	public function render_fullpp_content( $attributes, $content ) {
+		// Bail out if current user is admin or the author.
+		if ( current_user_can( 'manage_options' ) || ( get_the_author_meta( 'ID' ) === get_current_user_id() ) ) {
+			return $content;
+		}
+
+		if (
+			! is_user_logged_in() ||
+			( is_user_logged_in() && ! up_get_user_ethereum_address() )
+		) {
+			$login_button_text       = up_get_general_settings( 'login_button_text', __( 'Login with Unlock', 'unlock-protocol' ) );
```

Comment:
Let's move this to its own function `render_login_content`... or even better, let's just reuse what the rest of the plugin already does?